cedar-policy/src/api.rs

```diff
@@ -360,7 +362,7 @@ pub struct Diagnostics {
     /// If no policies applied to the request, this set will be empty.
     reason: HashSet<PolicyId>,
     /// list of error messages which occurred
-    errors: HashSet<String>,
+    errors: Vec<EvaluationError>,
```

Comment:
line 33 contains pub use auth..
new change ->  errors: Vec<AuthorizationError>

Reply:
Oops, missed line 33. That's fine then.

Since the `is_authorized` function in `cedar-policy-core` returns an `AuthorizationError`, we'll also want to use that here, instead of an `EvaluationError`.